src/sentry/static/sentry/app/views/settings/components/dataPrivacyRules/rulesList.tsx

```diff
@@ -1,96 +1,214 @@
 };
 
-const RulesList = React.forwardRef<HTMLUListElement, Props>(function RulesList(
-  {rules, onShowEditRuleModal, onDeleteRule, disabled},
-  ref
-) {
-  return (
-    <List ref={ref} isDisabled={disabled}>
-      {rules.map(({id, method, type, source}) => {
-        const methodLabel = getMethodTypeLabel(method);
-        const typelabel = getRuleTypeLabel(type);
-        return (
-          <ListItem key={id}>
-            <TextOverflow>
-              {`[${methodLabel.label}] [${typelabel}] ${t('from')} [${source}]`}
-            </TextOverflow>
-            {onShowEditRuleModal && (
-              <Button
-                label={t('Edit Rule')}
-                size="small"
-                onClick={onShowEditRuleModal(id)}
-                icon={<IconEdit />}
-                disabled={disabled}
-              />
-            )}
-            {onDeleteRule && (
-              <Button
-                label={t('Delete Rule')}
-                size="small"
-                onClick={onDeleteRule(id)}
-                icon={<IconDelete />}
-                disabled={disabled}
-              />
-            )}
-          </ListItem>
-        );
-      })}
-    </List>
-  );
-});
-
-RulesList.propTypes = {
-  rules: PropTypes.array.isRequired,
-  onShowEditRuleModal: PropTypes.func,
-  onDeleteRule: PropTypes.func,
-  disabled: PropTypes.bool,
+type State = {
+  columnQtd: number;
+  hoveredClassname?: string;
 };
 
+class RulesList extends React.PureComponent<Props, State> {
+  state: State = {columnQtd: DEFAULT_COLUMN_QUANTITY};
+
+  componentDidMount() {
+    this.loadColumnsQuantity();
+  }
+
+  loadColumnsQuantity = () => {
+    let extraColumnQtd = 0;
+
+    if (this.props.onDeleteRule) {
+      extraColumnQtd += 1;
+    }
+    if (this.props.onShowEditRuleModal) {
+      extraColumnQtd += 1;
+    }
+
+    this.setState(prevState => ({
+      columnQtd: prevState.columnQtd + extraColumnQtd,
+    }));
+  };
+
+  handleMouseOver = (className: string) => () => {
+    this.setState({
+      hoveredClassname: className,
+    });
+  };
+
+  handleMouseLeave = () => {
+    this.setState({
+      hoveredClassname: undefined,
+    });
+  };
+
+  render() {
+    const {forwardRef, disabled, onDeleteRule, onShowEditRuleModal, rules} = this.props;
+    const {hoveredClassname, columnQtd} = this.state;
+
+    return (
+      <Grid
+        ref={forwardRef}
+        isDisabled={disabled}
+        hoveredClassname={hoveredClassname}
+        columnQtd={columnQtd}
+      >
+        {rules.map(({id, method, type, source, customRegularExpression}) => {
+          const className = `gridCell-${id}`;
+          const methodLabel = getMethodTypeLabel(method);
+          const typeLabel = getRuleTypeLabel(type);
+          const methodDescription =
+            type === RuleType.PATTERN ? customRegularExpression : typeLabel;
+
+          const gridCellProps = !disabled
+            ? {
+                onMouseOver: this.handleMouseOver(className),
```

Comment:
Do you want mouseEnter here instead? the mouseover event is also triggered when children are hovered.